pkg/neonvm/ipam/allocate.go

```diff
@@ -1,38 +1,54 @@
 package ipam
 
 import (
-	"context"
-	"fmt"
 	"net"
 
 	whereaboutsallocate "github.com/k8snetworkplumbingwg/whereabouts/pkg/allocate"
 	whereaboutslogging "github.com/k8snetworkplumbingwg/whereabouts/pkg/logging"
 	whereaboutstypes "github.com/k8snetworkplumbingwg/whereabouts/pkg/types"
+
+	"k8s.io/apimachinery/pkg/types"
 )
 
+type ipamAction = func(
+	ipRange RangeConfiguration,
+	reservation []whereaboutstypes.IPReservation,
+) (net.IPNet, []whereaboutstypes.IPReservation, error)
+
+// makeAcquireAction creates a callback which changes IPPool state to include a new IP reservation.
+func makeAcquireAction(vmName types.NamespacedName) ipamAction {
+	return func(ipRange RangeConfiguration, reservation []whereaboutstypes.IPReservation) (net.IPNet, []whereaboutstypes.IPReservation, error) {
+		return doAcquire(ipRange, reservation, vmName)
+	}
+}
+
+// makeReleaseAction creates a callback which changes IPPool state to deallocate an IP reservation.
```

Comment:
These comments are great - succinct, implies how to use it while only telling you what it does :)